core/src/main/java/com/linecorp/armeria/server/ExceptionReportingServerErrorHandler.java

```diff
@@ -0,0 +1,113 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.LongAdder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.server.logging.LoggingService;
+
+import io.netty.channel.EventLoopGroup;
+
+/**
+ * Implementation of {@link ServerErrorHandler} that is used to decorate other ServerErrorHandlers.
+ * When services are not annotated with {@link LoggingService}, exceptions are not caught be default.
+ * {@link ExceptionReportingServerErrorHandler} is used to catch those uncaught exceptions.
+ */
+class ExceptionReportingServerErrorHandler implements ServerErrorHandler {
+    private static final Logger logger = LoggerFactory.getLogger(ExceptionReportingServerErrorHandler.class);
+    private final ServerErrorHandler delegate;
+    private final LongAdder counter;
+    private final long intervalInSeconds;
+    private boolean isScheduled;
+    @Nullable
+    private Throwable lastThrownException;
+    @Nullable
+    private ScheduledFuture<?> reportExceptionsSchedule;
+
+    ExceptionReportingServerErrorHandler(ServerErrorHandler serverErrorHandler, long intervalInSeconds) {
+        delegate = serverErrorHandler;
+        this.intervalInSeconds = intervalInSeconds;
+        counter = new LongAdder();
+    }
+
+    /**
+     * If it's required to log exceptions, increase the counter and store last thrown exception.
+     */
+    @Nullable
+    @Override
+    public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable cause) {
+        if (isScheduled && ctx.shouldLogException()) {
+            lastThrownException = cause;
+            counter.increment();
+        }
+        return delegate.onServiceException(ctx, cause);
+    }
+
+    @Nullable
+    @Override
+    public AggregatedHttpResponse renderStatus(ServiceConfig config, @Nullable RequestHeaders headers,
+                                               HttpStatus status, @Nullable String description,
+                                               @Nullable Throwable cause) {
+        return delegate.renderStatus(config, headers, status, description, cause);
+    }
+
+    /**
+     * Starts logging exceptions.
+     * @param workerGroup eventLoopGroup on which logging exceptions will be scheduled.
+     */
+    void start(EventLoopGroup workerGroup) {
+        requireNonNull(workerGroup, "workerGroup");
+        if (!isScheduled && intervalInSeconds > 0) {
```

Comment:
We could remove the `interval > 0` part once we add the assertion in the constructor.

Reply:
Yes ~ added some validation on the constructor to check whether the duration is non-positive.